tokio/src/runtime/io/registration.rs

```diff
@@ -219,11 +219,17 @@ impl Registration {
         loop {
             let event = self.readiness(interest).await?;
 
+            let coop = crate::future::poll_fn(crate::runtime::coop::poll_proceed).await;
+
             match f() {
                 Err(ref e) if e.kind() == io::ErrorKind::WouldBlock => {
                     self.clear_readiness(event);
                 }
-                x => return x,
+                x => {
+                    coop.made_progress();
+
+                    return x;
```

Comment:
This affects only things that use `async_io`. I believe that we discussed a different fix on one of the issues, which is the following:

 * Polling for readiness returns Pending on missing coop budget, but does not consume budget.
 * IO operations, even non-async ones, consume a budget, but do not check it.

With the above change, `async_io` would get the right behavior automatically.

Reply:
This feels like the correct way to do the accounting, but I'm worried it might lead to more significant changes in behavior than we may intend (e.g many users might be expecting try_ ops to not touch the buget).

Perhaps we can use this current method for the patch releases, and then the other method for the next minor version release we do?

Reply:
That works for me.